City-of-Helsinki/parking-permits-admin-ui
# PV-257: Add CookieHub widget (cookie consent)

## Description

Add CookieHub widget for cookie consent management, similar to [hel.fi](https://hel.fi/).

## Context

Due to GDPR shenanigans, we need to ask for cookie consents. This is handled by CookieHub, which provides an embeddable widget to handle this for us. Shows up in the bottom of the page. 

## How has this been tested?

Manually, in a local development environment. by using a personal CookieHub account with the starter (free) tier. No unit tests.

The translations/styles are customized inside the CookieHub dashboard, so my personal version differs from the one used in [hel.fi](https://hel.fi/). CookieHub's also based on domains that it can actually access, so unfortunately I can't verify whether it works or not 100% in localhost. At least the widget renders correctly and doesn't show any errors, so it _should_ work, but definitely needs to be verified in the test environment with a real domain.

## Manual testing instructions for reviewers

- There's a new dependency, `react-helmet`, so make sure you run `yarn install` in your Docker container.
- Create a CookieHub account and get a valid widget URL from the dashboard... or ask for mine, but either way, not providing the URL for it in the PR
- Update your `.env.development.local` file:
  - `REACT_APP_COOKIEHUB_URL="https://cookiehub.net/c2/xxxxxxxx.js"`, replace value with a valid URL
- `docker compose up`
- Navigate to the site, verify that at the bottom of the page there's a dialog asking for cookie consents.
- Also verify that clicking "Settings" (if English, if in other languages, figure it out) opens up another dialog and that the dialog disappears in the bottom of the page if you save the settings
- Refresh the page, verify that the other buttons in the bottom dialog ("Allow all", "Reject all") closes the dialog.

As mentioned earlier, the configurations for the widget's content comes from CookieHub itself, so expect it to *not* be 1:1 with the one in [hel.fi](https://hel.fi/).

## Screenshots

![image](https://user-images.githubusercontent.com/2333857/189371324-aef8b0f8-a20c-49df-a1e4-97912e81153d.png)
Dialog in the bottom.

![image](https://user-images.githubusercontent.com/2333857/189371402-71acad72-7512-463f-a1af-19561d4e94ee.png)
Settings dialog after clicking "Settings"
